.travis.yml

```diff
@@ -0,0 +1,13 @@
+language: go
+go:
+  - 1.5
+before_install:
+  - sudo apt-get update
+  - sudo apt-get install -qq libgl1-mesa-dev xorg-dev
+install:
+  - go get golang.org/x/tools/cmd/vet
+script:
+  - go get -t -v ./v3.1/...
+  - diff -u <(echo -n) <(gofmt -d -s .)
```

Comment:
I thought this was a very common idiom, that's why there's no comment. E.g., see [here](https://github.com/go-gl/examples/blob/26d37bbf6265a393404a126a43ac69930370e1fb/.travis.yml), [here](https://github.com/sourcegraph/go-diff/blob/7eefa0faf57871a70d7a2c368a454593f07797bc/.travis.yml#L18) and [here](https://github.com/shurcooL/goe/blob/3bcc0e08dda8182005ab103f77e7bf8749d1cf7e/.travis.yml#L8).

It tests that there are no `gofmt` issues.

More specifically, it prints a unified diff (for ease of reading) between empty string (`echo -n`) and the result of running `gofmt -d -s .`, which runs `gofmt` on current directory (and [therefore](https://golang.org/cmd/gofmt/) all .go files inside, recursively) and prints any `-d`ifferences from correct `gofmt` formatting. It also checks for `-s`implify opportunities.

So, if there are no `gofmt` issues, the diff is non-existent and exit code from `diff` is 0, build passes. If there are non-zero differences, then `diff` returns non-zero exit code. Running `gofmt -d` alone does not return a non-zero exit code if there are formatting issues.